scripts/travisci/check_flake8.sh

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env bash
+# Exit status of this script
+exit_status=0
+
+# Add an option to run the incremental verification locally to reduce the time
+# spent in travis for error verification. It's required to rebase the feature
+# branch with the master branch to avoid checking on unknown changes.
+while getopts ":l" opt; do
+  case ${opt} in
+    l)
+      TRAVIS_PULL_REQUEST="false"
+      ;;
+    \?)
+      echo "usage: check_flake8 [-l]"
+      echo "    -l    run incremental verification locally"
+      exit 1
+      ;;
+  esac
+done
+
+# ABSOLUTE VERIFICATION:
+# The following errors are analyzed in all python files across the repository.
+#
+# TABS
+# W191 indentation contains tabs
+# E101 indentation contains mixed spaces and tabs
+errors="E101,W191,"
+
+# WHITESPACE IN EXPRESSION AND STATEMENTS
+# E201 whitespace after '(', '[' or '{'
+# E202 whitespace before ')', ']' or '}'
+errors+="E201,E202,"
+# E241 multiple spaces after ','
+errors+="E241,"
+# E211 whitespace before '(' or '['
+errors+="E211"
+# E221 multiple spaces before operator
+errors+="E221,"
+# W291 trailing whitespace
+errors+="W291"
+# E702 multiple statements on one line (semicolon)
+errors+="E702"
+
+# PROGRAMMING RECOMMENDATIONS
+# E714 use is not operator rather than not ... is.
+errors+="E714"
+
+find -iname "*.py" | flake8 --select=${errors}
+if [[ $? -ne 0 ]]; then
+  exit_status=1
+fi
+
+
+# INCREMENTAL VERIFICATION:
+# The following errors are analyzed only in the modified code in the pull
+# request branch.
+#
+# INDENTATION
+# E125 continuation line with same indent as next logical line
+# E128 continuation line under-indented for visual indent
+errors="E125,E128,"
```

Comment:
IMO this would be much more readable as an array
```sh
errors=(
# INDENTATION
E125  # continuation line with same indent as next logical line
E128  # continuation line under-indented for visual indent
# MAXIMUM LINE LENGTH
E501  # line too long
# etc
)

# stuff

flake8 --diff "${quotes}" --select="${errors[@]}" --ignore="${ignore[@]}"
```

Reply:
Done.